lib/Dialect/SV/Transforms/SVExtractTestCode.cpp

```diff
@@ -626,17 +619,32 @@ void SVExtractTestCodeImplPass::runOnOperation() {
         continue;
       }
 
+      SmallPtrSet<Operation *, 32> opsToErase;
       bool anyThingExtracted = false;
       anyThingExtracted |= doModule(rtlmod, isAssert, "_assert", assertDir,
-                                    assertBindFile, bindTable);
+                                    assertBindFile, bindTable, opsToErase);
       anyThingExtracted |= doModule(rtlmod, isAssume, "_assume", assumeDir,
-                                    assumeBindFile, bindTable);
+                                    assumeBindFile, bindTable, opsToErase);
       anyThingExtracted |= doModule(rtlmod, isCover, "_cover", coverDir,
-                                    coverBindFile, bindTable);
+                                    coverBindFile, bindTable, opsToErase);
 
       // Inline any modules that only have inputs for test code.
       if (anyThingExtracted)
-        inlineInputOnly(rtlmod, instanceGraph, bindTable);
+        inlineInputOnly(rtlmod, instanceGraph, bindTable, opsToErase);
+
+      // Erase any instances that were extracted, and their forward dataflow.
+      // Also erase old instances that were inlined and can now be cleaned up.
+      // Parts of the forward dataflow may have been nested under other ops to
+      // erase, so keep track of what's being erased as we go to avoid erasing
+      // children of ops that were already erased.
+      SmallPtrSet<Operation *, 32> erasedOps;
+      for (auto *op : opsToErase) {
+        if (erasedOps.contains(op))
+          continue;
+        op->walk([&](Operation *erasedOp) { erasedOps.insert(erasedOp); });
```

Comment:
Does `op->erase()` erase child operations? If so I feel `erarsedOps` contains freed operations and will cause UAF when calling `erasedOps.contains`.

Reply:
hmm good question. it does erase child ops, which is the problem i'm trying to avoid. let me think about a different solution.